lib/matplotlib/backend_managers.py

```diff
@@ -74,6 +72,44 @@ def __init__(self, canvas):
         self.keypresslock = widgets.LockDraw()
         self.messagelock = widgets.LockDraw()
 
+        self._figure = None
+        if figure:
+            self.set_figure(figure)
+
+    @property
+    def canvas(self):
+        """Canvas managed by FigureManager"""
+        return self._figure.canvas
+
+    @property
+    def figure(self):
+        """Figure that holds the canvas"""
+        return self._figure
+
+    @figure.setter
+    def figure(self, figure):
+        self.set_figure(figure)
+
+    def set_figure(self, figure, update_tools=True):
```

Comment:
``figure=None`` to match the class init method?  Not 100% sure on this though, thinking of someone doing an array iteration calling this method and would want a value like this to mean no-change, but I err towards the feeling that they should handle that themselves.